source/extensions/filters/common/fault/fault_config.cc

```diff
@@ -10,6 +11,35 @@ namespace Filters {
 namespace Common {
 namespace Fault {
 
+FaultAbortConfig::FaultAbortConfig(
+    const envoy::extensions::filters::http::fault::v3::FaultAbort& abort_config)
+    : percentage_(abort_config.percentage()) {
+  switch (abort_config.error_type_case()) {
+  case envoy::extensions::filters::http::fault::v3::FaultAbort::ErrorTypeCase::kHttpStatus:
+    provider_ = std::make_unique<FixedAbortProvider>(abort_config.http_status());
+    break;
+  case envoy::extensions::filters::http::fault::v3::FaultAbort::ErrorTypeCase::kHeaderAbort:
+    provider_ = std::make_unique<HeaderAbortProvider>();
+    break;
+  case envoy::extensions::filters::http::fault::v3::FaultAbort::ErrorTypeCase::ERROR_TYPE_NOT_SET:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+}
+
+absl::optional<Http::Code>
+FaultAbortConfig::HeaderAbortProvider::status_code(const Http::HeaderEntry* header) const {
+  if (header == nullptr) {
+    return absl::nullopt;
+  }
+
+  uint64_t value;
+  if (!absl::SimpleAtoi(header->value().getStringView(), &value)) {
+    return absl::nullopt;
+  }
+
+  return static_cast<Http::Code>(value);
```

Comment:
Should you check if in the range 100 to 600 here and return nullopt if not? I would probably actually do 200 to 600 to avoid informational headers which might break things in strange ways, but up to you.